### PR TITLE
feat(sdk/go): add kubernetes authentication provider

### PR DESCRIPTION
**Pull request by GeorgeMac**
This adds a new `AuthenticationProvider` which automatically configures the client to use our Kubernetes authentication method.
With this provider configured, the client automatically authenticates using the Kubernetes method, by exchanging the service account token on the host with the target Flipt instance. It uses the default service account token path, which can be overridden. While the previously exchanged client token exists and is valid (not expired) within some configurable leeway, it uses the previously exchanged client token.
When the token draws close to expiry it exchanged it automatically for a new one. It does so by re-reading the service account token and exchanging with Flipt again. This is important, because Kubernetes will update the contents of that file, as its own service account tokens are automatically refreshed.

This PR additionally introduces k8s auth method to the matrix of methods used in the integration tests.
Doing so allows us to ensure the Go SDK works across the entire API with this authentication method.
To achieve this, there is a little thin JWKS provider implementation added that supports using its own CA.
This is to mimic how Kubernetes runs its own OIDC discovery mechanisms in-cluster.

### CODE REVIEW

File: internal/cmd/protoc-gen-go-flipt-sdk/main.go [hunk 1]
```diff
@@ -303,10 +313,96 @@ func (p StaticTokenAuthenticationProvider) Authentication() (string, error) {
...
+}
+
+// WithKubernetesExpiryLeeway configures the duration leeway for deciding when to refresh
+// the client token. The default is 10 seconds, which ensures that tokens are automatically refreshed
+// when their is less that 10 seconds of lifetime left on the previously fetched client token.
+func WithKubernetesExpiryLeeway(d time.Duration) KubernetesAuthenticationProviderOption {
+	return func(kctp *KubernetesAuthenticationProvider) {
+		kctp.leeway = d
+	}
+}
```
Comment (yquansah) [inline internal/cmd/protoc-gen-go-flipt-sdk/main.go:354]: I wonder if we should have restrictions for the value of this `leeway`?

Comment (GeorgeMac) [reply internal/cmd/protoc-gen-go-flipt-sdk/main.go:354]: I think we might be ok with just a sensible default like it has here. I hope no one really has to mess with this.
They can't really do much damage by setting it innapropriately either. At worst they will request a new token on every request (if leeway > duration of lifetime), or they may encounter the odd expired error if they set it too low.
I think any bounds we pick here might be arbitrary, im not sure.

File: sdk/go/doc.go [hunk 2]
```diff
@@ -33,23 +33,37 @@ The following is an example of creating an instance of the SDK using the HTTP tr
...
 		provider := sdk.JWTAuthenticationProvider("some-flipt-jwt")
 		client := sdk.New(transport, sdk.WithAuthenticationProvider(provider))
 	}
 
+- [KubernetesAuthenticationProvider]:
+
+This automatically uses the service account token on the host and exchanges it with Flipt for a Flipt client token credential. The credential is then used to authenticate requests, again via the Authentication header and the Bearer scheme. It ensures that the client token is not-expired and requests fresh tokens automatically without intervention. Use this method to automatically authenticate your application with a Flipt deployed into the same Kubernetes cluster.
+
+	func main() {
+	    provider := sdk.NewKuberntesAuthenticationProvider(transport)
```
Comment (markphelps) [inline sdk/go/doc.go:63]:
```suggestion
	    provider := sdk.NewKubernetesAuthenticationProvider(transport)
```

Comment (markphelps) [reply sdk/go/doc.go:63]: also can we link to the docs for each of these authentication methods?

**Review by markphelps — COMMENTED** [review]
looking great!

**Comment by GeorgeMac** [comment]
Something funny going on in actions land. The env var we generate for use in naming the upload artefacts is occassionally going missing 🤔 . Some builds it gets generated appropriately, sometimes not. I think something funny is going on in the actions environment.

**Comment by GeorgeMac** [comment]
OK, I see this only appears to happen when the test itself fails because it steps over the env var setting.

File: build/testing/integration.go [hunk 3]
```diff
@@ -697,3 +730,117 @@ func signJWT(key crypto.PrivateKey, claims interface{}) string {
...
+					"name": "integration-test-7d26f049-kdurb",
+					"uid":  "bd8299f9-c50f-4b76-af33-9d8e3ef2b850",
+				},
+				"serviceaccount": map[string]any{
+					"name": "integration-test",
+					"uid":  "4f18914e-f276-44b2-aebd-27db1d8f8def",
+				},
+			},
+		}), nil
+}
```
Comment (yquansah) [inline build/testing/integration.go:846]: The function name is pretty explanatory as far as what it is doing, but if you don't mind, could you provide a brief explanation in your mind for what is going on here (in a comment)?

**Review by yquansah — APPROVED** [review]
Looks great!

File: build/testing/integration.go (hunk 3 above)
Comment (GeorgeMac) [reply build/testing/integration.go:846]: That is a great shout, it is chunky.

**Review by markphelps — APPROVED** [review]
looks great! mind updating the docs as well?

**Pull request merged**